internal/cmd/base/flags.go

```diff
@@ -581,25 +586,53 @@ func (f *FlagSet) StringSliceVar(i *StringSliceVar) {
 		Usage:      i.Usage,
 		Default:    def,
 		EnvVar:     i.EnvVar,
-		Value:      newStringSliceValue(initial, i.Target, i.Hidden),
+		Value:      newStringSliceValue(initial, i.Target, i.Hidden, i.NullCheck),
 		Completion: i.Completion,
 	})
 }
 
 type stringSliceValue struct {
-	hidden bool
-	target *[]string
+	hidden    bool
+	nullCheck func() bool
+	target    *[]string
 }
 
-func newStringSliceValue(def []string, target *[]string, hidden bool) *stringSliceValue {
+func newStringSliceValue(def []string, target *[]string, hidden bool, nullCheck func() bool) *stringSliceValue {
 	*target = def
 	return &stringSliceValue{
-		hidden: hidden,
-		target: target,
+		hidden:    hidden,
+		nullCheck: nullCheck,
+		target:    target,
 	}
 }
 
 func (s *stringSliceValue) Set(val string) error {
+	trimmedVal := strings.TrimSpace(val)
+	// Don't enable this behavior if nullCheck is not defined
+	if s.nullCheck != nil {
+		// If we got null in, go through some checks
+		if trimmedVal == "null" {
+			// Ensure null is allowed here
+			if !s.nullCheck() {
+				return fmt.Errorf(`"null" is not an allowed value`)
+			}
+			// If we have at least one value already and it's not "null" then
+			// error; we don't check if _all_ are "null" since presumably this
+			// function prevents there being more values than just "null" if
+			// that's specified
+			if len(*s.target) > 0 && (*s.target)[0] != "null" {
+				return fmt.Errorf(`"null" cannot be combined with other values`)
+			}
+			// Set the target to only contain "null" and return
+			*s.target = []string{"null"}
+			return nil
+			// Something came in that isn't "null" but we already have "null"
```

Comment:
nit: can we move this comment back one tab or start it on a new line? (unless this way is already an established precedent)

Reply:
Yeah, I had done that and didn't notice that autoformatting pulled it forward. I moved it down into the if clause itself.